tests/test_with_children.py

```diff
@@ -0,0 +1,154 @@
+            "<div><h1>My title</h1><p></p></div>",
+        ),
+        (
+            my_component(title="My title")["My ", "content"],
+            "<div><h1>My title</h1><p>My content</p></div>",
+        ),
+    ],
+)
+def test_with_children_str(component: h.Renderable, expected: str) -> None:
+    assert str(component) == expected
+
+
+@pytest.mark.parametrize(
+    ("component", "expected"),
+    [
+        (
+            my_component,
+            "<div><h1>Default title</h1><p></p></div>",
+        ),
+        (
+            my_component["My ", "content"],
+            "<div><h1>Default title</h1><p>My content</p></div>",
+        ),
+        (
+            my_component(title="My title"),
+            "<div><h1>My title</h1><p></p></div>",
+        ),
+        (
+            my_component(title="My title")["My ", "content"],
+            "<div><h1>My title</h1><p>My content</p></div>",
+        ),
+    ],
+)
+def test_with_children_html(component: h.Renderable, expected: str) -> None:
+    assert component.__html__() == expected
+
+
+@pytest.mark.parametrize(
+    ("component", "expected"),
+    [
+        (
+            my_component,
+            b"<div><h1>Default title</h1><p></p></div>",
+        ),
+        (
+            my_component["My ", "content"],
+            b"<div><h1>Default title</h1><p>My content</p></div>",
+        ),
+        (
+            my_component(title="My title"),
+            b"<div><h1>My title</h1><p></p></div>",
+        ),
+        (
+            my_component(title="My title")["My ", "content"],
+            b"<div><h1>My title</h1><p>My content</p></div>",
+        ),
+    ],
+)
+def test_with_children_encode(component: h.Renderable, expected: bytes) -> None:
+    assert component.encode() == expected
+
+
+@pytest.mark.parametrize(
+    ("component", "expected"),
+    [
+        (
+            my_component,
+            ["<div>", "<h1>", "Default title", "</h1>", "<p>", "</p>", "</div>"],
+        ),
+        (
+            my_component["My ", "content"],
+            [
+                "<div>",
+                "<h1>",
+                "Default title",
+                "</h1>",
+                "<p>",
+                "My ",
+                "content",
+                "</p>",
+                "</div>",
+            ],
+        ),
+        (
+            my_component(title="My title"),
+            ["<div>", "<h1>", "My title", "</h1>", "<p>", "</p>", "</div>"],
+        ),
+        (
+            my_component(title="My title")["My ", "content"],
+            [
+                "<div>",
+                "<h1>",
+                "My title",
+                "</h1>",
+                "<p>",
+                "My ",
+                "content",
+                "</p>",
+                "</div>",
+            ],
+        ),
+    ],
+)
+def test_with_children_iter_chunks(component: h.Renderable, expected: list[str]) -> None:
+    assert list(component.iter_chunks()) == expected
```

Comment:
All tests that check str/encode/iter_chunks could be added in test_render.py. It would then check all the variants. It would also use the render() fixture which would also test async versions eventually in #38.

Reply:
Nice, didn't know about those! I've ported the tests to use the existing infrastructure in `test_render.py` now.